lib/arke/orderbook.rb

```diff
@@ -1,10 +1,38 @@
-# encoding: UTF-8
-# frozen_string_literal: true
-
-require 'rbtree'
-require_relative 'price_level'
-
 module Arke
   class Orderbook
+
+    attr_reader :book
+
+    def initialize(market)
+      @market = market
+      @book = {
+        sell: [],
+        buy: []
+      }
+    end
+
+    def add(order)
+      @book[order.side].push(order) unless @book[order.side].find {|b| b.id == order.id }
+    end
+
+    def remove(order)
+      index = @book[order.side].index {|b| b.id == order.id }
+      @book[order.side].delete_at(index) unless index.nil?
+    end
+
+    def update(order)
+      index = @book[order.side].index {|b| b.id == order.id }
+      @book[order.side][index] = order unless index.nil?
+    end
+
+    def contains?(order)
+      @book[order.side].find {|b| b.id == order.id } ? true : false
```

Comment:
you return true if true, so @book[order.side].find { |b| b.id == order.id } should work as well (and a little space before |b|)

Reply:
book[order.side].find { |b| b.id == order.id } returns nil or found element